pkg/agent/agent.go

```diff
@@ -89,9 +98,54 @@ func (a *Agent) Run(ctx context.Context) error {
 		nodeAttestor = cat.GetNodeAttestor()
 	}
 
-	as, err := a.attest(ctx, sto, cat, metrics, nodeAttestor)
-	if err != nil {
-		return err
+	var as *node_attestor.AttestationResult
+
+	if a.c.RetryBootstrap {
+		attBackoffClock := clock.New()
+		attBackoff := backoff.NewBackoff(
+			attBackoffClock,
+			bootstrapBackoffInterval,
+			backoff.WithMaxElapsedTime(bootstrapBackoffMaxElapsedTime),
+		)
+
+		for {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			default:
+			}
```

Comment:
Is this really needed?

Reply:
Revisiting this code, I think it is not needed. I was overthinking. Thanks!